lib/floe/workflow.rb

```diff
@@ -110,9 +108,12 @@ def run_nonblock
     def step_nonblock
       return Errno::EPERM if end?
 
+      start_state unless context.state_started?
       result = current_state.run_nonblock!(context)
       return result if result != 0
 
+      end_state
```

Comment:
Yea, `result != 0` means the state has not finished yet. Something like a `Wait` or `Task` that has not completed yet.

In the future, it will call into `run_nonblock`

But I think I'll take this out and revisit what I'm trying to do here. Thanks

Reply:
oh i see it's the count of incomplete ones?  Interesting - Maybe result is just a bad variable name and something like "remaining_something" would be clearer.